synapse/lib/ast.py

```diff
@@ -24,6 +26,40 @@ def __init__(self, item=None):
 class StormBreak(StormCtrlFlow): pass
 class StormContinue(StormCtrlFlow): pass
 
+@s_cache.memoize()
+def getGlobRegex(text):
+    '''
+    Compute a tag regex for glob style matching.
+
+    Args:
+        text (str): String to generate a regex for.
+
+    Notes:
+        This function acts as a global cache for compiled regex objects for tags.
+
+    Returns:
+        regex.Regex: Compiled tag regex.
+    '''
+    # Glob helpers
+    glob_smark = '*'
+    glob_mmark = '**'
+    glob_sre = r'[^\.]+?'
+    glob_mre = '.+'
+
+    text = text.lower()
+
+    def _cmpGlobParts(s, sep='.'):
+        parts = s.split(sep)
+        parts = [p.replace(glob_mmark, glob_mre) for p in parts]
+        parts = [p.replace(glob_smark, glob_sre) for p in parts]
+        regex = '{}'.format(sep).join(parts)
```

Comment:
the sep needs to be escaped as well otherwise `foo.bar` would match `fooxbar`

Reply:
Better would be:
`parts = []
for part in s.split(sep):
    parts.append(regex.sub(r'(\\\*\\\*)|(\\\*)', lambda m: r'[^.]+?' if m.group(1) is None else
          ...: r'.+', regex.escape(part)))` 

(and probably precompile the re beforehand.)